kyaml/openapi/openapi.go

```diff
@@ -26,13 +34,45 @@ type ResourceSchema struct {
 // as metadata, replicas and spec.template.spec
 func SchemaForResourceType(t yaml.TypeMeta) *ResourceSchema {
 	initSchema()
-	rs, found := schemaByResourceType[t]
+	rs, found := c.schemaByResourceType[t]
 	if !found {
 		return nil
 	}
 	return &ResourceSchema{Schema: rs}
 }
 
+// AddSchema parses s, and adds definitions from s to the global schema.
+func AddSchema(s []byte) error {
+	if c.schemaByResourceType == nil {
+		c.schemaByResourceType = map[yaml.TypeMeta]*spec.Schema{}
+	}
+	return parse(s)
+}
+
+// GetSchema parses s into a ResourceSchema, resolving References within the
+// global schema.
+func GetSchema(s string) (*ResourceSchema, error) {
+	var sc spec.Schema
+	if err := sc.UnmarshalJSON([]byte(s)); err != nil {
+		return nil, errors.Wrap(err)
+	}
+	if sc.Ref.String() != "" {
+		r, err := resolve(rootSchema(), &sc.Ref)
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
+		sc = *r
+	}
+
+	return &ResourceSchema{Schema: &sc}, nil
+}
+
+// NoUseBuiltInSchema can be called with b=false to prevent using the built-in Kubernetes
```

Comment:
passing bools is hard to understand at the call point

how about no setter, and just a noarg method like `suppressBuiltinSchemaUse()` (or the opposite)

Reply:
done